GliaWidgets/Sources/ViewController/Common/Alert/AlertViewController.swift

```diff
@@ -27,10 +27,28 @@ class AlertViewController: UIViewController {
             accepted: () -> Void,
             declined: () -> Void
         )
+
+        /// Indicating whether an alert is as a result on user action,
+        /// like pressing "End" engagement button, or pressing stop screen sharing button.
+        /// Based on that user-initiated can be only the `confirmation`case
+        fileprivate var isUserInitiated: Bool {
+            switch self {
+            case .confirmation:
+                return true
+            case .message, .singleAction, .singleMediaUpgrade, .screenShareOffer:
+                return false
+            }
+        }
     }
 
     let viewFactory: ViewFactory
 
+    /// Indicating whether alert can be replaced with another alert.
+    /// When `kind.isUserInitiated` is true, the alert should not be dismissed
+    var canBeReplaced: Bool {
```

Comment:
I wonder if there's a better wording for this. `isReplaceable`, `allowsSubstitution`?

Reply:
958dc160f6820f72e126ecfb07bd21edb6fe3f49